framework/helpers/BaseArrayHelper.php

```diff
@@ -532,6 +532,9 @@ public static function index($array, $key, $groups = [])
     public static function getColumn($array, $name, $keepKeys = true)
     {
         $result = [];
+        if(!is_array($array)) {
```

Comment:
This is potential BC break & may start yelding silent errors. User should ensure supplied $array is array.

Reply:
Nope its not, BC, because, if user must dont always like this
If (is_array($array) || is_itterable($array) {
$columns = ArrayHelper::getColumn.. 
}
He always must do that check because getColumn run foreach and dont ensures that $array is really iterrable object, i think its not good.

Reply:
If getColumn expects $array = array or itterable object we must change signature of getColumn.
Now its like getColumn expects 'something' and try to itterate over it, its not good.

Reply:
Personally I'm a fan of sanity checks, but this is going to give me a result that I don't expect.   If you're not able to iterate over the array, this method should throw an exception not a null array.

Reply:
> Personally I'm a fan of sanity checks, but this is going to give me a result that I don't expect. If you're not able to iterate over the $array variable because it's not an array or iterable, this method should throw an exception not a null array.

Yes i agree with you, but not this method dont throw Exception.
This method just try to itterate on variable.